### PR TITLE
single log line 

### PR DESCRIPTION
**Pull request by Seetaramayya**

### CODE REVIEW

**Comment by jordi133** [comment]
Approved for merge after changing type alias Monitor to AlertRuleActorRef and consistent use of sender()

File: src/it/scala/performanceanalysis/LogSubmissionTest.scala [hunk 1]
```diff
@@ -14,18 +17,42 @@ class LogSubmissionTest extends IntegrationTestBase with ScalaFutures with Twitt
       When("I do a HTTP GET to '/' on the LogReceiver port")
       val request  = http.Request(http.Method.Get, "/")
       val response = performLogReceiverRequest(request)
-      whenReady(response, timeout(1.seconds)) { result =>
-        assert(result.getStatusCode() === 405)
-        Then("the response should have statuscode 405")
+      whenReady(response, timeout(1 second)) { result =>
+        result.getStatusCode() shouldBe 405
```
Comment (ruud) [inline src/it/scala/performanceanalysis/LogSubmissionTest.scala]: for consistency:
405 should be changed into StatusCodes.MethodNotAllowed

File: src/it/scala/performanceanalysis/LogSubmissionTest.scala [hunk 2]
```diff
@@ -14,18 +17,42 @@ class LogSubmissionTest extends IntegrationTestBase with ScalaFutures with Twitt
...
+        Then("the response should have status code 405")
```
Comment (ruud) [inline src/it/scala/performanceanalysis/LogSubmissionTest.scala:22]: for consistency:
405 should be changed into StatusCodes.MethodNotAllowed

File: src/main/scala/performanceanalysis/logreceiver/LogReceiverActor.scala [hunk 3]
```diff
@@ -13,28 +13,31 @@ object LogReceiverActor {
...
-  def receive: Receive = normal(Map.empty[String, ActorRef])
-
-  def normal(logParserActors: Map[String, ActorRef]): Receive = {
-    case SubmitLogs(componentId, logs) =>
-      handleSubmitLogs(logParserActors, componentId, logs)
-    case RegisterNewLogParser(componentName, newLogParser) =>
-      handleNewLogParser(logParserActors, componentName, newLogParser)
+  private type ComponentId = String
+  private type LogParser = ActorRef
+  private type LogParsers = Map[ComponentId, LogParser]
```
Comment (ruud) [inline src/main/scala/performanceanalysis/logreceiver/LogReceiverActor.scala]: As before, maybe its not a good idea to create a type for a Map. The name suggests a List or Set.

File: src/main/scala/performanceanalysis/server/Protocol.scala [hunk 4]
```diff
@@ -60,7 +65,7 @@ object Protocol {
   case class Details(metrics: List[Metric])
 
   /**
-    * Used by Administrator towards LogReceiver to notify it of a new LogReceiver actor
+    * Used by Administrator towards LogReceiver to notify it ofa new LogReceiver actor
```
Comment (ruud) [inline src/main/scala/performanceanalysis/server/Protocol.scala]: of a is ok, isn't it ;)

**Comment by ruud** [comment]
Just some minor comments from me.
+1 for merging pull-request

File: src/it/scala/performanceanalysis/LogSubmissionTest.scala (hunk 1 above)
Comment (Seetaramayya) [inline src/it/scala/performanceanalysis/LogSubmissionTest.scala]: I agree

File: src/main/scala/performanceanalysis/server/Protocol.scala (hunk 4 above)
Comment (Seetaramayya) [inline src/main/scala/performanceanalysis/server/Protocol.scala]: Indeed, I think `intellij` did it. I'll change.

File: src/it/scala/performanceanalysis/LogSubmissionTest.scala (hunk 2 above)
Comment (Seetaramayya) [inline src/it/scala/performanceanalysis/LogSubmissionTest.scala:22]: I agree, I'll change

File: src/main/scala/performanceanalysis/logreceiver/LogReceiverActor.scala (hunk 3 above)
Comment (Seetaramayya) [inline src/main/scala/performanceanalysis/logreceiver/LogReceiverActor.scala]: I'll change

Comment (jordi133) [inline src/main/scala/performanceanalysis/logreceiver/LogReceiverActor.scala]: Agree, `LogParsersById` is a bit more clear (and prevents further discussion on this ;)

Comment (Seetaramayya) [inline src/main/scala/performanceanalysis/logreceiver/LogReceiverActor.scala]: :+1:

Comment (ruud) [inline src/main/scala/performanceanalysis/logreceiver/LogReceiverActor.scala]: Not creating a new type would be consistent with the rest

**Comment by Seetaramayya** [comment]
@ruud Did you push changes by mistake? Or Am I missing the point?

**Pull request merged**